### PR TITLE
Add Redis storage for orchestrator 

### PR DESCRIPTION
**Pull request by iuwqyir**
### TL;DR

Added Redis support for orchestrator storage.

### What changed?

- Introduced Redis configuration options in `cmd/root.go`
- Added `RedisConfig` struct in `configs/config.go`
- Implemented `RedisConnector` in `internal/storage/redis.go`
- Updated `NewConnector` function to support Redis in `internal/storage/connector.go`
- Modified RPC method support check in `internal/common/rpc.go`
- Added Redis-related dependencies in `go.mod` and `go.sum`

### How to test?

1. Configure Redis connection details using the new flags:
   - `--storage-orchestrator-redis-poolSize`
   - `--storage-orchestrator-redis-addr`
   - `--storage-orchestrator-redis-password`
   - `--storage-orchestrator-redis-db`
2. Run the application with Redis storage enabled
3. Verify that the orchestrator can store and retrieve data using Redis
4. Test the functionality of `GetLatestPolledBlockNumber`, `StoreLatestPolledBlockNumber`, `GetBlockFailures`, `StoreBlockFailures`, and `DeleteBlockFailures` methods

### CODE REVIEW

File: internal/common/rpc.go [hunk 1]
```diff
@@ -95,17 +95,17 @@ func (rpc *RPC) checkSupportedMethods() error {
 	var getLogsResult interface{}
 	logsErr := rpc.RPCClient.Call(&getLogsResult, "eth_getLogs", map[string]string{"fromBlock": "0x0", "toBlock": "0x0"})
 	if logsErr != nil {
-		return fmt.Errorf("eth_getBlockByNumber method not supported: %v", logsErr)
+		return fmt.Errorf("eth_getLogs method not supported: %v", logsErr)
 	}
 	log.Debug().Msg("eth_getLogs method supported")
 
+	var traceBlockResult interface{}
```
Comment (AmineAfia) [inline internal/common/rpc.go:102]: curious why you check for this even when tracing is disabled

Comment (iuwqyir) [reply internal/common/rpc.go:102]: I am not. It is making a trace request only if configuration allows it. 
But if it's allowed we still need to check if the rpc allows it so we make a request.
That's why we can't solely rely on the config

**Review by AmineAfia — APPROVED** [review]
